### ISSUE TITLE
Application crash trying to read log file in /var/log

### ISSUE THREAD

**Issue by ckulkarni-cribl** [Ravi-D/Log-Collection-Simple-REST-API#1]
Hi Ravi, thank you so much for submitting the assignment.

I'm trying to run the application on mac and reading files in /var/log. reading a simple 19MB file is crashing the process and making swagger-ui unresponsive. I dont see any stack traces or log files in the application directory, any idea what could be going on?

<img width="1419" alt="image" src="https://user-images.githubusercontent.com/106271079/196527414-1497b29b-7d37-4307-96aa-7b82da30ad5c.png">

This happens consistently - regardless of Release/Debug configuration and running from visual studio/command line.

**Comment by Ravi-D** [comment]
Hey, thanks for checking! Would it be possible for you to share this file with me? I can try to process it on a Mac and set a couple breakpoints to see what's going on. @ckulkarni-cribl

**Comment by ckulkarni-cribl** [comment]
Sure! here's the file i was trying - 
[install.log](https://github.com/Ravi-D/Log-Collection-Simple-REST-API/files/9814649/install.log)

fwiw - api worked fine with smaller files (~20kb)

**Comment by Ravi-D** [comment]
Oh that's good to know at least lol. In a flurry of defects at least it does work to an extent. 

Gotcha. I think this directly correlates to your latest post about StreamReader performance. I will prioritize that story and this one above the others for now.

**Comment by Ravi-D** [comment]
![image](https://user-images.githubusercontent.com/13991686/196570887-7afb020d-967d-4a07-b6fd-f643c91748dc.png)

**Root cause**
It is two fold, Swagger itself appears to be the _main_ culprit.
1. Swagger appears unable to output large amount of lines anyways. Not sure what the exact "cutoff" of printable lines is.
2. The existing file reading strategy (IO.File library) was also adding more time to the process than I wanted. After all that talk about StreamReader I used File.ReadLines instead of StreamReader.ReadLine! 😲 


**Testing**
Postman
1. Reversing the results of IO.File.ReadLines -> 3 second response time.
2. StreamReader, added valid lines to a List, reverse iterated the list, added to resultBuilder and returned -> screenshot above, <300ms response time. Way more in line to what I was expecting. 

When debugging in either file reading strategy, every line of install.log was still correctly output to the console quite quickly. The hanging happens _after_ the Http method returns ContentResult back to Swagger.

**Resolution**
So I used the StreamReader AND tried it on Postman instead (this run was uncached), and the contents of the file are coming back way faster. (Screenshot above)

**another future idea**
I am still considering a custom "ReverseStreamReader" class in the future to avoid filling and reverse iterating a List, but any possible gain there will surely be negligible compared to the huge speed boost already gained by ditching File.ReadLines(fullPath).Reverse()

Latest PR should be up soon.
